core/src/ibc/connection_03/types.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use rlp;
+use rlp::{DecoderError, Rlp, RlpStream};
+
+#[repr(u8)]
+#[derive(Copy, Clone)]
+pub enum ConnectionState {
+    INIT = 0,
+    TRYOPEN = 1,
+    OPEN = 2,
+}
+
+impl rlp::Encodable for ConnectionState {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.append(&(*self as u8));
+    }
+}
+
+// FIXME: current commitment_23::Prefix is too generic.
+pub type CommitmentPrefix = String;
+pub type Identifier = String;
+
+pub struct ConnectionEnd {
+    pub state: ConnectionState,
+    pub counterparty_connection_identifier: Identifier,
+    pub counterparty_prefix: CommitmentPrefix,
+    pub client_identifier: Identifier,
+    pub counterparty_client_identifier: Identifier,
+    // FIXME: implement version
+}
+
+impl rlp::Encodable for ConnectionEnd {
+    fn rlp_append(&self, s: &mut rlp::RlpStream) {
```

Comment:
Did you check that RLP **macro** can handle this?

Reply:
Sure, I'll check them.

Reply:
I checked this code: https://github.com/CodeChain-io/rlp/blob/64fdbc5758e05483f62fae99521520f566eaca9d/rlp-derive/src/en.rs#L21
It seems it is suitable to use it. I'll try it.

Reply:
I changed it. Could you check it again?